src/grammar.lalrpop

```diff
@@ -561,6 +562,45 @@ ArgNames: ArgNames = {
     <label:VarName> <name:DiscardName> => ArgNames::LabelledDiscard { label, name },
 }
 
+RecordUpdate: UntypedExpr = {
+    // <s:@L> <m:VarName> "#" <n:UpName> <args:RecordUpdateArgs> <e:@L> => UntypedExpr::RecordUpdate {
+    //     location: location(s, e),
+    //     module: Some(m),
+    //     name: n,
+    //     spread: args.0,
+    //     args: args.1,
+    // },
+
+    <s:@L> "#" <n:UpName> <args:RecordUpdateArgs> <e:@L> => UntypedExpr::RecordUpdate {
```

Comment:
Similarly, I'd prefer not having the `"#"` in front, but without it, the grammar is ambiguous because we'd need 2 tokens of lookahead to determine whether a spread is present, and disambiguate record updates, record construction, etc.

This is again a case that we should be able to handle, but I wasn't sure what the best way of doing so is.